packages/gatsby-source-drupal/src/normalize.js

```diff
@@ -12,11 +13,37 @@ const getHref = link => {
 
 exports.getHref = getHref
 
-const nodeFromData = (datum, createNodeId, entityReferenceRevisions = []) => {
+const nodeFromData = async (
+  datum,
+  createNodeId,
+  entityReferenceRevisions = [],
+  pluginOptions
+) => {
   const { attributes: { id: attributeId, ...attributes } = {} } = datum
   const preservedId =
     typeof attributeId !== `undefined` ? { _attributes_id: attributeId } : {}
   const langcode = attributes.langcode || `und`
+  const type = datum.type.replace(/-|__|:|\.|\s/g, `_`)
+
+  const isFile = isFileNode(datum, type)
+
+  const url = isFile
+    ? pluginOptions.baseUrl + getFileUrl(datum.attributes)
+    : null
+
+  const imageSize = isFile ? await probeImageSize(url) : {}
```

Comment:
Can we put this in a queue? I'd guess this will slow down sourcing a lot unless I get parallelize it.